File: bench/realworld/bench_json.go
// bench_json.go - parse an 18 MB JSON array, ITERS times, summing value fields.
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"time"
)

// All four fields are decoded, so Go materialises the same data as every other
// language here. Omitting them would let encoding/json skip most of the record
// and would not be measuring the same workload.
type Rec struct {
	Id     int64  `json:"id"`
	Name   string `json:"name"`
	Value  int64  `json:"value"`
	Active bool   `json:"active"`
}

func main() {
	data, _ := os.ReadFile("bench_data.json")
	iters := 5
	t0 := time.Now()
	var total int64 = 0
	for k := 0; k < iters; k++ {
		var arr []Rec
		json.Unmarshal(data, &arr)
		for i := range arr {
			total += arr[i].Value
		}
	}
	elapsed := time.Since(t0).Milliseconds()
	fmt.Printf("result: %d\n", total)
	fmt.Printf("elapsed: %d ms\n", elapsed)
}

File: bench/realworld/bench_json.py
# bench_json.py - parse an 18 MB JSON array, ITERS times, summing value fields.
import json, time
with open("bench_data.json") as f:
    text = f.read()
ITERS = 5
t0 = time.monotonic()
total = 0
for _ in range(ITERS):
    arr = json.loads(text)
    for rec in arr:
        total += rec["value"]
elapsed = int((time.monotonic() - t0) * 1000)
print(f"result: {total}")
print(f"elapsed: {elapsed} ms")
